chanzuckerberg/single-cell-data-portal
# feat: implement improved rollup backend

## Reason for Change

- #6400
- Implementing a new rollup algorithm as part of the performance improvement epic.

## Changes

- Replaced the existing rollup algorithm with a simpler version that leverages pandas explode and groupby to perform the rollup.
- Make some QOL improvements to the performance profiling workflow

## Testing steps

- Unit tests updated
- Rdev inspected to confirm equivalence of data in the frontend.
- performance profiling was done on this branch vs #6427 (which is a copy of `main`). 
[scatter_plots_20240108_132010.pdf](https://github.com/chanzuckerberg/single-cell-data-portal/files/13866359/scatter_plots_20240108_132010.pdf)
